CodeStarterCamp_Week5/STEP1.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  File.swift
+//  CodeStarterCamp_Week5
+//
+//  Created by 김경준 on 2022/11/14.
+//
+
+import Foundation
+// MARK: - STEP1. 야곰 엔터 오디션
+
+class Person {
+    let name: String
+    let height: Int
+    
+    init(name: String, height: Int) {
+        self.name = name
+        self.height = height
+    }
+}
+
+enum Level {
+    case A
+    case B
+    case C
+}
+
+protocol Talent {
+    var singing: Level { get }
+    var dancing: Level { get }
+    var acting: Level { get }
+}
+
+protocol BadPersonality {
+    var frequancyOfCursing: Level { get }
+}
+
+class TalentedPerson: Person, Talent {
+    let singing: Level
+    let dancing: Level
+    let acting: Level
+    
+    init(name: String, height: Int, singing: Level, dancing: Level, acting: Level) {
+        self.singing = singing
+        self.dancing = dancing
+        self.acting = acting
+        super.init(name: name, height: height)
+    }
+}
+
+class TalentedPersonWithBadPersonality: Person, Talent, BadPersonality {
+    let singing: Level
+    let dancing: Level
+    let acting: Level
+    let frequancyOfCursing: Level
+    
+    init(name: String, height: Int, singing: Level, dancing: Level, acting: Level,
+         frequancyOfCursing: Level) {
+        self.singing = singing
+        self.dancing = dancing
+        self.acting = acting
+        self.frequancyOfCursing = frequancyOfCursing
+        super.init(name: name, height: height)
+    }
+}
+
+struct AuditionManager {
+    var totalApplicantsList: [Person]
+    var passedApplicantsList: [Person] = []
+    
+    mutating func appendPassedList(applicant: Person) {
+        self.passedApplicantsList.append(applicant)
+    }
+    
+    mutating func removeFromPassedList(applicant: Person) {
+        self.passedApplicantsList.removeLast()
+    }
+    
+    mutating func checkTalent(applicant: Person) {
+        if let talentedApplicant = applicant as? Talent {
+            if talentedApplicant.singing == Level.A {
+                appendPassedList(applicant: applicant)
+            } else if talentedApplicant.acting == Level.A {
+                appendPassedList(applicant: applicant)
+            } else if talentedApplicant.dancing == Level.A {
+                appendPassedList(applicant: applicant)
+            }
+        }
+    }
+    
+    mutating func checkBadPersonality(applicant: Person) {
+        if (applicant is BadPersonality) == true {
+            self.removeFromPassedList(applicant: applicant)
+        }
+    }
```

Comment:
```suggestion
    mutating func checkBadPersonality(applicant: Person) {
        if applicant is BadPersonality {
            self.removeFromPassedList(applicant: applicant)
        }
    }
```

이렇게만 조건을 주어도 true / false 로 나뉘지 않을까요 ? 당연히 true 일 때만 if 문 안으로 들어 올 것입니다 ☺️